### ISSUE TITLE
Creating a dynamic assembly with the same name as any assembly in the process causes debugging to fail within that assembly

### ISSUE THREAD

**Issue by jacob-mink-1996** [dotnet/roslyn#79612]
**Version Used**: 
4.14.0-3.25326.11

**Steps to Reproduce**:

1. Debug an EXE in which a dynamic assembly is created with the same name as an assembly loaded into that process
2. Observe that upon creation of the assembly, the debugger exhibits internal compiler error. 

```csharp
using System.Reflection;
using System.Reflection.Emit;

var (x, y) = (10, 20);
var asm = AssemblyBuilder.DefineDynamicAssembly(Assembly.GetExecutingAssembly().GetName(), AssemblyBuilderAccess.Run);
Console.WriteLine($"{x} + {y} = {x + y}");
```
Set a breakpoint on the `var asm` line, observe `x` and `y` beforehand, step over it, and observe the error.

I know this is a bit of a hysterical situation, but we shot ourselves with this foot-gun and spent a good long while inspecting things until we discovered the Modules window showing something like
"The real dll.dll" | Debuggable | PDB Path |
"The real dll" | Not Debuggable | No PDB Path |
where "The real dll" was the dynamic one.

Since it breaks the debugger, we thought to report rather than just live with our fix (obviously, slap a GUID on there).

**Expected Behavior**:
1. (Preferred) The debugger finds another/the next assembly to debug in (maybe not y'all, but this reports as a compiler error in VS).
2. .NET should disallow the creation of a dynamic assembly with the same name as another in the process (not y'all, but still... good to note).

**Actual Behavior**:
Internal compiler error in the debugger Auto/Local/Immediate windows

**Comment by jjonescz** [comment]
> Observe that upon creation of the assembly, the debugger exhibits internal compiler error.

Can you be more specific about what error you see? Thanks.

**Comment by jacob-mink-1996** [comment]
@jjonescz thanks for jumping in. Weirdly, the top level error is just that: "Internal error in the C# compiler".

<img width="1943" height="137" alt="Image" src="https://github.com/user-attachments/assets/6d3d88bd-58b1-4f92-a564-b1dbcd6a6596" />

Inspecting gets me a little more:

<img width="1402" height="126" alt="Image" src="https://github.com/user-attachments/assets/e50e0faf-1ad8-4ca4-815a-cef35e965539" />

And here's that stack trace:
```
Microsoft.CodeAnalysis.ExpressionEvaluator.ExpressionCompiler.dll!Microsoft.CodeAnalysis.ExpressionEvaluator.ExpressionCompiler.Microsoft.VisualStudio.Debugger.ComponentInterfaces.IDkmClrExpressionCompiler.GetClrLocalVariableQuery(DkmInspectionContext, DkmClrInstructionAddress, Boolean)+0xC8
Microsoft.VisualStudio.Debugger.Engine.dll!Microsoft.VisualStudio.Debugger.EntryPoint.IDkmClrExpressionCompiler_GetClrLocalVariableQuery(IntPtr, IntPtr, IntPtr, Byte, IntPtr&)+0x31
mscorlib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()+0xC
Microsoft.VisualStudio.Debugger.Engine.dll!XapiExceptionProcessing.ThrowHR(Int32, Boolean)+0x14
Microsoft.VisualStudio.Debugger.Engine.dll!Microsoft.VisualStudio.Debugger.Evaluation.DkmInspectionContext.GetClrLocalVariableQuery(DkmClrInstructionAddress, Boolean, String&, DkmCompiledClrLocalsQuery&)+0x58
vsdebugeng.manimpl.dll!VSDebugEngine.ManagedEE.LocalsProvider.Create(DkmWorkList, DkmInspectionContext, DkmStackWalkFrame, Boolean, Action`1)+0x30
```

You should be able to see it from the 5-liner above, too, if you get the chance to try the code.
